feat: LEAP-1709: Add region index displaying to relations list

Gondragos opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follow the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] The best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
<img width="753" alt="image" src="https://github.com/user-attachments/assets/e677ea9c-a323-4879-9fd2-17bd915417e0" />

1. There is no region indexes indication for the relations panel at all.
2. There is an issue with setting region indexes when the region tab is not active.



#### What is the new behavior?
<img width="753" alt="image" src="https://github.com/user-attachments/assets/1de7bc11-9c7b-4fdb-9377-86151b449efc" />

1. Indexes are set every time when a region is created or deleted.
2. Indexes are updated on sort / filter state changes.
3. There is an indication of region index in the relations panel.
4. Styles are slightly adjusted.



### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [x] integration
- [ ] unit



### Which logical domain(s) does this change affect?
`Relations Panel`, `region_index`

Gondragos committed:
fix: LEAP-1709: Add region index display and improve styling

Added a new region index element in the details panel with corresponding styles for alignment and visibility. Updated margin and layout adjustments for consistent design across elements. Introduced gap styling in relations panel for better spacing of node elements.

Gondragos committed:
fix: LEAP-1709: Ensure consistent region indexing across the app

Region indices are now set consistently during sorting and creation. Removed redundant region index assignment in OutlinerTree to avoid discrepancies.

Gondragos committed:
Stylelint fixes

Gondragos committed:
fix: LEAP-1709: Refactor region indexes updating, resolve crash issue

Refactored region index updates to a dedicated method `updateIndexes` for better consistency and maintainability. This ensures indexes are recalculated correctly after sorting, creation, or deletion of regions.

Resolved a problem causing crashes due to observer side effects.

Gondragos committed:
test: Add region index tests and enhance LSF test helpers

Added new utility methods for handling regions, labels, and localStorage in LSF helpers. Introduced customizable callbacks for `init` and improved functionality for ordering and deleting regions in the Sidebar.

Gondragos committed:
fix: optimize region index management and appearance updates

Replaced `setRegionIndex` logic with a computed `region_index` derived from a new `regionIndexMap`. Add some new triggers for updateAppearenceFromState.

Gondragos committed:
fix: Update region index handling for non-real regions

Removed redundant updateIndexes calls and adjusted region_index getter to handle non-real regions properly. This improves performance and prevents errors related to invalid indexing.

Gondragos committed:
fix: handle string values in localStorage setup

Ensure localStorage values are correctly serialized only when necessary. This prevents potential issues when dealing with string data types.

Gondragos committed:
test: Ensure region consistency during delete/create operations

Add integration test to verify region list consistency when regions are deleted or created. This includes handling changes affecting the entire list and ensuring correct relabeling and ordering.

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/12950305767)
> Successfully merged:  create mode 100644 web/tsconfig.json

Gondragos committed:
Merge branch 'develop' into 'fb-leap-1709/region-index-in-relations'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/12950305767

Gondragos committed:
fix: remove debug logs from Hotkeys helper

bmartel commented on web/libs/editor/src/stores/Annotation/Annotation.js:
```diff
@@ -656,14 +656,17 @@ const _Annotation = types
       }
     },
+    updateAppearenceFromState() {
+      self.areas.forEach((area) => area.updateAppearenceFromState && area.updateAppearenceFromState());
```
```suggestion
      self.areas.forEach((area) => area.updateAppearenceFromState?.());
```

Commit:
Refactoring

Co-authored-by: bmartel <brandonmartel@gmail.com>

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13003755788)
> Successfully merged:  delete mode 100644 docs/themes/v2/source/images/prompts/example_subset2.png

Gondragos committed:
Merge branch 'develop' into 'fb-leap-1709/region-index-in-relations'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13003755788

Gondragos committed:
fix: ensure annotation appearance updates on  sort/filter change

Previously, annotation appearance updates were missing during certain interactions. This fix ensures `updateAppearenceFromState` is called after sorting or grouping regions. Added an integration test to verify label consistency when changing order.

Gondragos committed:
Merge branch 'fb-leap-1709/region-index-in-relations' of ssh://github.com/heartexlabs/label-studio into fb-leap-1709/region-index-in-relations

Gondragos committed:
fix: handle potential null annotation in AreaMixin

Ensure `updateAppearenceFromState` is called only if `annotation` is defined. This prevents runtime errors during the destruction phase.

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13034169739)
> Successfully merged:  delete mode 100644 web/tsconfig.json

Gondragos committed:
Merge branch 'develop' into 'fb-leap-1709/region-index-in-relations'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13034169739

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13035009123)
> Successfully merged: Already up to date.

Commit:
Empty commit to rebuild

Gondragos committed:
fix: add optional chaining to prevent runtime errors

Gondragos committed:
Merge remote-tracking branch 'origin/fb-leap-1709/region-index-in-relations' into fb-leap-1709/region-index-in-relations

Gondragos committed:
Add comments

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13118365392)
> Successfully merged:  create mode 100644 label_studio/tests/sdk/test_export.py

Gondragos committed:
Merge branch 'develop' into 'fb-leap-1709/region-index-in-relations'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13118365392

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13123442144)
> Successfully merged:  5 files changed, 47 insertions(+), 8 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-leap-1709/region-index-in-relations'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13123442144

MihajloHoma commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13130457692)
> Successfully merged:  1 file changed, 27 insertions(+)

MihajloHoma committed:
Merge branch 'develop' into 'fb-leap-1709/region-index-in-relations'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13130457692
